### ISSUE TITLE
Suggest use of negative array indices for modern bash (v4.1 or later)

### ISSUE THREAD

**Issue by Tamaranch** [koalaman/shellcheck#1667]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
declare -a a=('a' 'b')
echo "${a[${#a[@]}-1]}"

```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

^-- SC????: You could use negative indices: `echo "${a[-1]}"`
